cli/commandhandler.h

```diff
@@ -56,11 +60,109 @@ void handleCreateSignedTransactionWithPemFile(cxxopts::ParseResult const &result
+    auto const gasPrice = result["gas-price"].as<uint64_t>();
+    auto const nonce = result["nonce"].as<uint64_t>();
+
+    auto const canFreeze = result["can-freeze"].as<bool>();
+    auto const canWipe = result["can-wipe"].as<bool>();
+    auto const canPause = result["can-pause"].as<bool>();
+    auto const canMint = result["can-mint"].as<bool>();
+    auto const canBurn = result["can-burn"].as<bool>();
+    auto const canChangeOwner = result["can-change-owner"].as<bool>();
+    auto const canUpgrade = result["can-upgrade"].as<bool>();
+    auto const canAddSpecialRoles = result["can-add-roles"].as<bool>();
+
+    auto const pemPath = result["pem"].as<std::string>();
+
+    PemFileReader const pemFileReader(pemPath);
+    Address const sender = pemFileReader.getAddress();
+    Address const receiver = Address(config.issueESDTSCAddress);
+
+    Transaction tx;
+    tx.m_nonce = nonce;
+    tx.m_sender = std::make_shared<Address>(sender);
+    tx.m_receiver = std::make_shared<Address>(receiver);
+    tx.m_gasPrice = gasPrice;
+    tx.m_chainID = config.chainID;
+
+    ESDTProperties const esdtProperties
+            {canFreeze,
+             canWipe,
+             canPause,
+             canMint,
+             canBurn,
+             canChangeOwner,
+             canUpgrade,
+             canAddSpecialRoles};
+
+    prepareTransactionForESDTIssuance(tx, token, ticker, supply, decimals, esdtProperties);
+    internal::signTransaction(tx, pemFileReader);
+
+    ProxyProvider proxy(config.proxyUrl);
+    auto const txHash =  proxy.send(tx).hash;
+    std::cerr<< "Transaction hash: " << txHash << "\n";
+}
+
+void handleTransferESDT(cxxopts::ParseResult const &result)
+{
+    auto const config = NetworkConfig().config();
+
+    auto const nonce = result["nonce"].as<uint64_t>();
+    auto const gasPrice = result["gas-price"].as<uint64_t>();
+    auto const gasLimit = result["gas-limit"].as<uint64_t>();
+    auto const receiverAdr = result["receiver"].as<std::string>();
+    auto const token = result["token"].as<std::string>();
+    auto const value = result["value"].as<std::string>();
+    auto const function = result["function"].as<std::string>();
+    auto const args = result["args"].as<std::vector<std::string>>();
+
+    auto const pemPath = result["pem"].as<std::string>();
+
+    PemFileReader const pemFileReader(pemPath);
+    Address const sender = pemFileReader.getAddress();
+    Address const receiver = Address(receiverAdr);
+
+    Transaction tx;
+    tx.m_nonce = nonce;
+    tx.m_value = value;
+    tx.m_sender = std::make_shared<Address>(sender);
+    tx.m_receiver = std::make_shared<Address>(receiver);
+    tx.m_gasPrice = gasPrice;
+    tx.m_gasLimit = gasLimit;
+    tx.m_chainID = config.chainID;
+
+    prepareTransactionForESDTTransfer(tx, token, function);
+
+    if (!args.empty())
+    {
+        std::string txData (tx.m_data->begin(), tx.m_data->end());
+        for (auto const &arg : args)
```

Comment:
Not sure, I think the arguments concatenation / serialization logic is written in other place as well. Perhaps we can extract it and keep it in a single place (in a future PR)?

Reply:
Similar functionality is written in `SCArguments` class

```c++

    SCArguments args;

    BigUInt const bigUInt("10"); //saves internally hex value of decimal 10  = 0a
    Address const address("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"); //saves internally hex value of public key = "0139472eff6886771a982f3083da5d421f24c29181e63888228dc81ca60d69e1"
    std::string const str("foo"); //saves internally hex representation of ASCII string "foo" = "666f6f"

    args.add(bigUInt);
    args.add(address);
    args.add(str);
 
    // This will return the concatenated arguments with "@" delimiter
    args.asOnData() == "@0a@0139472eff6886771a982f3083da5d421f24c29181e63888228dc81ca60d69e1@666f6f
```

**However**, for CLI we can **NOT** deduce (in a nice, clean way) the type of the argument. For instance, if CLI --args option looks like this:

```bash
erdcpp esdt transfer --function=someFunction --args=100,erd1j
```

- Was `100` meant to be an ASCII string or a BigUint? 
- Was `erd1j` meant to be a bech32 addres or an ASCII string?

So we just ask the user to fill the **already hex encoded** arguments, so here I add them directly at the end of the data, without the help of `SCArguments`.